tests/rbd/test_performance_immutable_cache.py

```diff
@@ -0,0 +1,236 @@
+"""Test case covered -
+    CEPH-83574134 - Configure immutable object cache daemon
+    and validate client RBD objects
+
+    Pre-requisites :
+    1. Cluster must be up and running with capacity to create pool
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files
+
+    Test Case Flow:
+    1. Create RBD based pool and an Image
+    2. Enable the immutable cache client settings
+    3. Install the ceph-immutable-object-cache package
+    4. Create a unique Ceph user ID, the keyring
+    5. Enable the ceph-immutable-object-cache daemon with created client
+    6. Write some data to the image using FIO
+    7. Perform snapshot,protect and clone of rbd images
+    8. Read the cloned images from the cache path
+    9. Repeat the above operations without immutable cache
+    10.check the performance make sure cache gives good performnace
+"""
+import datetime
+import pdb
+import re
+
+from test_rbd_immutable_cache import configure_immutable_cache
+
+from ceph.rbd.initial_config import initial_rbd_config
+from ceph.rbd.utils import getdict, random_string
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.rbd import create_snap_and_clone
+from tests.rbd.krbd_io_handler import krbd_io_handler
+from utility.log import Log
+
+log = Log(__name__)
+
+immutable_cache_path = "/mnt/immutable-cache/"
+
+
+def compare_performance(time_with_cache, time_without_cache):
+    """
+    Compare the performance of write and read operations with and without cache.
+
+    Args:
+        time_with_cache (str): Time taken for write and read operations with cache.
+        time_without_cache (str): Time taken for write and read operations without cache.
+
+    Raises:
+        Exception: If the performance without cache is not better than with cache.
+    """
+    regex = r"(\d{1,2})m(\d{1,2}).(\d{1,3})s"
```

Comment:
Will you always get milliseconds as well? What if it was 0ms will it show 1m2.0s?
Because if it shows 1m2s this regex won't work.

Reply:
thanks for the notice yes it didn't work so created new comparision method itself.